libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -525,6 +527,50 @@ void AP_DDS_Client::update_topic(geographic_msgs_msg_GeoPointStamped& msg)
     }
 }
 
+bool AP_DDS_Client::update_topic(ardupilot_msgs_msg_Status& msg)
+{
+    // Fill the new message.
+    msg.vehicle_type = static_cast<uint8_t>(AP::fwversion().vehicle_type);
+    msg.armed = AP_Notify::flags.armed;
+    msg.mode = AP::vehicle()->get_mode();
+    msg.flying = AP_Notify::flags.flying;
+    uint8_t fs_iter = 0;
+    msg.failsafe_size = 0;
+    if ( AP_Notify::flags.failsafe_radio ) {
+        msg.failsafe[fs_iter++] = FS_RADIO;
+    }
+    if ( AP_Notify::flags.failsafe_battery ) {
```

Comment:
I wonder if we could get the battery failsafe state from the battery library.

For RC, EKF and GCS failsafe we probably need to do what you've done here

Reply:
I don't have an opinion about it, maybe we could port that limitation to a new issue?